File: backend/models/search_history.py
from models.database import db
from datetime import datetime

class SearchHistory(db.Model):
    __tablename__ = 'search_history'
    
    id = db.Column(db.Integer, primary_key=True)
    user_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)
    query = db.Column(db.Text, nullable=False)
    results_count = db.Column(db.Integer, default=0)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    
    def to_dict(self):
        return {
            'id': self.id,
            'query': self.query,
            'results_count': self.results_count,
            'created_at': self.created_at.isoformat() if self.created_at else None
        }
</create_file>

File: backend/JuriIQ.Core/Models/SearchHistory.cs
using System;

namespace JuriIQ.Core.Models
{
    public class SearchHistory
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public string Query { get; set; } = string.Empty;
        public int ResultsCount { get; set; }
        public string? Filters { get; set; }
        public DateTime SearchTime { get; set; } = DateTime.UtcNow;
    }
}